quickwit/quickwit-doc-mapper/src/default_doc_mapper/mapping_tree.rs

```diff
@@ -231,41 +238,76 @@ fn insert_json_val(
     doc_json.insert(last_field_name.to_string(), json_val);
 }
 
-trait NumVal: Sized + Into<TantivyValue> {
+trait NumVal: Sized + FromStr + ToString + Into<TantivyValue> {
     fn from_json_number(num: &serde_json::Number) -> Option<Self>;
 
-    fn from_json(json_val: JsonValue) -> Result<TantivyValue, String> {
-        if let JsonValue::Number(num_val) = json_val {
-            Ok(Self::from_json_number(&num_val)
+    fn from_json(json_val: JsonValue, coerce: bool) -> Result<TantivyValue, String> {
+        match json_val {
+            JsonValue::Number(num_val) => Self::from_json_number(&num_val)
+                .map(|num| num.into())
                 .ok_or_else(|| {
                     format!(
                         "Expected {}, got inconvertible JSON number `{}`.",
                         type_name::<Self>(),
                         num_val
                     )
-                })?
-                .into())
-        } else {
-            Err(format!("Expected JSON number, got `{json_val}`.",))
+                }),
+            JsonValue::String(str_val) if coerce => {
+                str_val.parse::<Self>().map(|num| num.into()).map_err(|_| {
+                    format!(
+                        "Expected {}, got inconvertible JSON string `{}`.",
+                        type_name::<Self>(),
+                        str_val,
+                    )
+                })
+            }
```

Comment:
It would be nice to improve discoverability of the feature as follows...
```
            JsonValue::String(str_val) => {
                let num_val_res: TantivyValue = str_val.parse().map(Self::into);
                if coerce {
                    //...
                } else {
                    if num_val_res.is_ok() {
                         return Err("some message mentioning the existence of the coerce parameter")
                    } else {
                         return Err("message explaining we expected a number")
                    }
                }
            }
```